astropy/coordinates/sites.py

```diff
@@ -0,0 +1,138 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Observatories accessible by the `sites` module originate from the IRAF
+Observatory Database, and are stored in ``data/observatories.json``.
+Longitudes are listed with positive to the West.
+
+Additions or corrections to the observatory list can be submitted via Pull
+Request to the [astropy-data GitHub repository](https://github.com/astropy/astropy),
+updating the ``location.json`` file.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import json
+from difflib import get_close_matches
+
+from ..utils.data import get_pkg_data_contents, get_file_contents
+from ..extern import six
+from .earth import EarthLocation
+
+__all__ = ['get_site', 'get_site_names']
+
+# Observatory database and list of names:
+_builtin_site_dict = {'_site_db_uninitialized': True}
+_builtin_site_names = []
+
+
+def _parse_sites_json(jsondb, names, sitedict):
+    for site in jsondb:
+        location = EarthLocation.from_geodetic(jsondb[site]['longitude'],
+                                               jsondb[site]['latitude'],
+                                               jsondb[site]['elevation_meters'])
+        names.append(jsondb[site]['name'])
+        for alias in jsondb[site]['aliases']:
+            sitedict[alias.lower()] = location
+
+def _get_builtin_sites():
+    """
+    Load observatory database from data/observatories.json and parse them into
+    a dictionary in memory.
+    """
+    if _builtin_site_dict.get('_site_db_uninitialized', False):
```

Comment:
Wouldn't it be easier to just test `if not _buildin_site_names:`?
With that the whole `_site_db_uninitialized` key stuff could be removed.